spec/ndx-holographic-stimulation.extensions.yaml

```diff
@@ -1,8 +1,132 @@
+    dtype: int8
+    doc: number of turns within a spiral
+- neurodata_type_def: TemporalFocusing
+  neurodata_type_inc: OptogeneticStimulusPattern
+  doc: table of parameters defining the temporal focusing beam-shaping
+  attributes:
+  - name: lateral_point_spread_function
+    dtype: text
+    doc: estimated lateral spatial profile or point spread function, expressed as
+      mean [um] ± s.d [um]
+  - name: axial_point_spread_function
+    dtype: text
+    doc: estimated axial spatial profile or point spread function, expressed as mean
+      [um]± s.d [um]
+- neurodata_type_def: PatternedOptogeneticStimulusSite
+  neurodata_type_inc: OptogeneticStimulusSite
+  doc: An extension of OptogeneticStimulusSite to include the geometrical representation
+    for the stimulus.
+  attributes:
+  - name: effector
+    dtype: text
+    doc: Light-activated effector protein expressed by the targeted cell (eg. ChR2)
+    required: false
+- neurodata_type_def: PatternedOptogeneticSeries
+  neurodata_type_inc: TimeSeries
+  doc: An extension of OptogeneticSeries to include the spatial patterns for the photostimulation.
+  attributes:
+  - name: unit
+    dtype: text
+    default_value: watts
+    doc: SI unit of data
+    required: false
+  datasets:
+  - name: data
+    dtype: numeric
+    dims:
+    - num_times
+    - num_rois
+    shape:
+    - null
+    - null
+    doc: The data values. The first dimension must be time. The second dimension represents
+      ROIs
+  - name: rois
+    neurodata_type_inc: DynamicTableRegion
+    dtype: int8
+    doc: references rows of ROI table
+  links:
+  - name: site
+    target_type: PatternedOptogeneticStimulusSite
+    doc: link to the patterned stimulus site
+  - name: stimulus_pattern
+    target_type: OptogeneticStimulusPattern
+    doc: link to the stimulus pattern
+    required: false
+  - name: device
+    target_type: Device
+    doc: link to the device used to generate the photostimulation
+  - name: spatial_light_modulator
+    target_type: SpatialLightModulator
+    doc: link to the spatial light modulator device
+    required: false
+  - name: light_source
+    target_type: LightSource
+    doc: link to the light source
+    required: false
+- neurodata_type_def: SpatialLightModulator
+  neurodata_type_inc: Device
+  doc: An extension of Device to include the Spatial Light Modulator metadata
+  attributes:
+  - name: model
+    dtype: text
+    doc: model of the Spatial Light Modulator, if known
+    required: false
+  - name: resolution
+    dtype: float32
+    doc: resolution of the Spatial Light Modulator in um, if known
+    required: false
+- neurodata_type_def: LightSource
+  neurodata_type_inc: Device
+  doc: An extension of Device to include the Light Sorce metadata
+  attributes:
+  - name: stimulation_wavelength
+    dtype: float32
+    doc: stimulation wavelength in nm
+  - name: filter_description
+    dtype: text
+    doc: filter_description
+    required: false
+  - name: power
```

Comment:
Should this be `peak_power`? Or `max_power`? Should it be on the `LightSource` device at all, or a property of the pattern or only the series data itself?

Trying to decouple as much potentially conflicting fields as possible; the Pinto lab for example had a device of `700 W` max power but they tuned it down to `400` for some stimulations

Reply:
I will leave peak_power in the LightSource as a property of the pulse laser (not required), and express the power of the stimulus in the data